cupy_backends/cuda/api/runtime.pxd

```diff
@@ -100,6 +99,321 @@ cdef extern from *:
     ctypedef struct IpcEventHandle 'cudaIpcEventHandle_t':
         unsigned char[64] reserved
 
+    ctypedef struct cudaUUID 'cudaUUID_t':
+        char bytes[16]
+
+    IF CUDA_VERSION >= 11000:
+        # We can't use IF in the middle of structs declaration
```

Comment:
We could use a single struct here, since Cython ignores the fields that are not present in the associated C one.
Also the order of the fields is not important, they don't have to match the C ones.

I decided to add all three structs to make this more explicit, but I can change it

Reply:
> We could use a single struct here, since Cython ignores the fields that are not present in the associated C one.
Also the order of the fields is not important, they don't have to match the C ones.

I didn't know that!

Reply:
> since Cython ignores the fields that are not present in the associated C one.

This is new to me too! I wonder how did they convince the compiler to pass...?

Reply:
Probably it's just wrapping the structure in another one and just re-mapping the fields? I didn't check the generated code